Reparameterizing an inline C model fails

**Partner issue of SasView/sasview#2958
** To see how to replicate issue in the SasView GUI, see above issue

**Describe the bug**
Reparameterizing a python model with inline C code will fail with an error in the format:

```
11:29:37 - root - ERROR - C:\Users\userDir\AppData\Local\miniconda3\envs\sasview_dev\Lib\site-packages\tinycc\tcc.exe -shared -rdynamic -Wall C:\Users\userDir\AppData\Local\Temp\1\sas64_reparameterizedModel_EF781090_2i9v5xc5.c -o C:\Users\userDir\.sasview\compiled_models\sas64_reparameterizedModel_EF781090.so
C:/Users/userDir/.sasview/plugin_models/oldModel.py:35: error: 'reparameterizedVariable' undeclared
```

on the developer version of SasView, where `reparameterizedModel` is the reparameterized file, `oldModel` is the model being reparameterized, and `reparameterizedVariable` is the name of a reparameterized variable accessed in inline C code within `oldModel`.

This bug occurs if any inline C code is included in the model, even if a separate C file is defined using `source = []` in the main python file. However, models that have defined C code entirely in a separate file are unaffected.

**Note: currently, this bug only affects lamellar.py in the 1.0.7 sasmodels version, but will also affect any user-created models with inline C unless resolved**

**To Reproduce**
Steps to reproduce the behavior:
1. Create a new user model.
2. Copy in the following code:
```
from numpy import inf

from sasmodels.core import reparameterize
from sasmodels.special import *

parameters = [
    # name, units, default, [min, max], type, description
	['new', '', 0, [-inf, inf], '', ''],
]

translation = """
    sld = new
"""

model_info = reparameterize('lamellar', parameters, translation, __file__)
```
3. The new model should return an error if a model check is run on it

**Expected behavior**
The model should have no issues being loaded or used with the new redefinitions.

**sasmodels version:**
 - Version: 1.0.7

**Suggested Solution:**
Either deprecate using inline C code (only one model currently uses it and it is easily replaceable), or fix the reparameterization code in sasmodels.core to support inline C code.

--- a/sasmodels/generate.py
+++ b/sasmodels/generate.py
@@ -977,22 +977,22 @@ def make_source(model_info):
     # one might for the volume of a sphere:
     #     form_volume="return M_4PI_3*cube(radius)"
     if isinstance(model_info.form_volume, str):
-        pars = call_table.form_volume_parameters
+        pars = base_table.form_volume_parameters
         source.append(_gen_fn(model_info, 'form_volume', pars))
     if isinstance(model_info.shell_volume, str):
-        pars = call_table.form_volume_parameters
+        pars = base_table.form_volume_parameters
         source.append(_gen_fn(model_info, 'shell_volume', pars))
     if isinstance(model_info.Iq, str):
-        pars = [q] + call_table.iq_parameters
+        pars = [q] + base_table.iq_parameters
         source.append(_gen_fn(model_info, 'Iq', pars))
     if isinstance(model_info.Iqxy, str):
-        pars = [qx, qy] + call_table.iq_parameters + call_table.orientation_parameters
+        pars = [qx, qy] + base_table.iq_parameters + base_table.orientation_parameters
         source.append(_gen_fn(model_info, 'Iqxy', pars))
     if isinstance(model_info.Iqac, str):
-        pars = [qab, qc] + call_table.iq_parameters
+        pars = [qab, qc] + base_table.iq_parameters
         source.append(_gen_fn(model_info, 'Iqac', pars))
     if isinstance(model_info.Iqabc, str):
-        pars = [qa, qb, qc] + call_table.iq_parameters
+        pars = [qa, qb, qc] + base_table.iq_parameters
         source.append(_gen_fn(model_info, 'Iqabc', pars))
 
     # Check for shell_volume function in source code

